tests/types/tillegaltyperecursion.nim

```diff
@@ -1,17 +0,0 @@
-discard """
-  cmd: "nim $target --threads:on $options $file"
-  errormsg: "illegal recursion in type 'TIRC'"
-  line: 12
-"""
-
-import net
-import strutils
-import os
-
-type
-    TIRC = object
-        Socket: Socket
-        Thread: Thread[TIRC]
-
-proc initIRC*(): TIRC =
-    result.Socket = socket()
```

Comment:
If the test should work, you probably need to modify the tests to ensure it continues to work instead of deleting the tests. And `initIRC` needs to be called to make sure it actually works.

Reply:
you can check the history, after latest modification, it doesn't really checked.